src/connection.ts

```diff
@@ -91,21 +91,25 @@ export class WebRTCConnection implements ic.Connection {
 
       log.trace(`${logPrefix} supported protocols - ${protocols}`);
 
-      let { stream, protocol } = await mshandle(rawStream, protocols, { signal: controller.signal });
-      if (metrics) {
-        metrics.trackStream({ stream, protocol, remotePeer: this.remotePeer });
-      }
+      try {
+        let { stream, protocol } = await mshandle(rawStream, protocols, { signal: controller.signal });
+        if (metrics) {
+          metrics.trackStream({ stream, protocol, remotePeer: this.remotePeer });
+        }
 
-      log.trace(`${logPrefix} handled protocol - ${protocol}`);
+        log.trace(`${logPrefix} handled protocol - ${protocol}`);
 
-      rawStream.stat.protocol = protocol;
-      let result = this.wrapMsStream(rawStream, stream);
+        rawStream.stat.protocol = protocol;
+        let result = this.wrapMsStream(rawStream, stream);
 
-      this.addStream(result);
+        this.addStream(result);
 
-      // handle stream
-      let { handler } = registrar.getHandler(protocol);
-      handler({ connection: this, stream: result });
+        // handle stream
+        let { handler } = registrar.getHandler(protocol);
+        handler({ connection: this, stream: result });
+      } catch (err) {
+        log.error('stream error: ', rawStream.id, rawStream.stat.direction);
```

Comment:
Do we really want to silence an error from this connection function? Will this not make it harder to use, as the user won't really be able to handle unexpected errors in this method cal?

Reply:
User does not call this method. This only sets up a callback handler for incoming streams.

Reply:
Even so, can we simply silence it? Does it not need handling in the flow where we might need to do something to act upon besides just logging? Or does the rest of the code flow handle this situation just the same as any?

Reply:
So, the possible errors in this try block could be as follows:
1. There is no protocol proposed in the incoming stream that we support. 
2. We select an incoming protocol, and there is a failure in the handler which is user-defined.
in either case, we do not want the error to affect the connection object, since the errors may not be network errors.

Reply:
I haven't looked too much into the business logic, but your reasoning seems sound. Thanks for the clarification. I'm still suspicious of the error silencing here, but I'll just keep it in the back of my mind for now. Thank you.